shoop/themes/classic_gray/views/product_preview.py

```diff
@@ -8,7 +8,21 @@
 from shoop.front.views.product import ProductDetailView
 
 
+class CGProductDetailView(ProductDetailView):
+    def get_context_data(self, **kwargs):
+        # By default the template rendering the basket add form
+        # uses the `request.path` as its' `next` value.
+        # This is fine if you are on product page but here in
+        # preview, we cannot redirect back to `/xtheme/product_preview`.
+
+        context = super(CGProductDetailView, self).get_context_data(**kwargs)
+        # Add `return_url` to context to avoid usage of `request.path` in
```

Comment:
because we are not in the product page the view that calls this is lives in url `/xtheme/product_preview` and it's called with ajax to the `/xtheme/products` which renders the modal containing the preview view. The preview view renders the same product price box the normal product detail page does, hence it has the `request.path` in `next`, which, in this case is the `/xtheme/product_preview` which is just a content for ajax request.